### DISCUSSION TITLE
tasks based on path/directory

### DISCUSSION THREAD

**Discussion by adriangalilea** [GothenburgBitFactory/taskwarrior#3364] [Q&A]
I want to set up taskwarrior in a way that the tasks I see are dependant on the path I'm navigating.

I tried several things, and I can't understand the taskwarrior way of doing things...

I made some tasks and created a project and a context for the project, I made a function on my .zshrc  that sets the taskwarrior context everytime I `cd` based on a dict {"/foo/bar" : "some_context"}, it didn't work perfect but it did the trick.

But then I opened 2 shells and worked in 2 different projects and the context of my first shell was hijacked when the context of my other shell was changed, I don't understand that default, but I'd need an alternative to solve my needs,

essentially I want to associate projects with paths and everytime I do `task` see the tasks associated to that project or all if none is associated to current path.

Any idea?

EDIT: Made some progress using:
- direnv
- defining an `.env` with TASKWARRIOR_PROJECT env inside for the appropriate project
- script does `vit project:${TASKWARRIOR_PROJECT}"`

I don't like that when I add a task is not appended to the current project, so I'll likely set the context too automatically, but this is hacks on top of hacks.

**Comment by jacostag** [comment]
I think that the issue is that you would have to re-enter/reload the proper directory so direnv can set the proper project.
When you switch to a different tab/shell/console it will it will keep the "original" project set by direnv, or a new one set by the new direnv.
You can set an alias that will set the project for you, something like:
```
tap = task add pro:${TASKWARRIOR_PROJECT}
tlp = task li pro:${TASKWARRIOR_PROJECT}
 ```

**Comment by jacostag** [comment]
@adriangalilea  I actually made a few scripts that will based on the directory, no alias or direnv needed or any other stuff. I am adding +dir tag and a directory or directories as annotations.
You can use that, still, not a project, but it avoids the context/project switching.
You can edit the script to show the task_ids for the current directory.
https://github.com/jacostag/tw_by_dir

**Reply by adriangalilea** [reply]
> @adriangalilea I actually made a few scripts that will based on the directory, no alias or direnv needed or any other stuff. I am adding +dir tag and a directory or directories as annotations. You can use that, still, not a project, but it avoids the context/project switching. You can edit the script to show the task_ids for the current directory. [jacostag/tw_by_dir](https://github.com/jacostag/tw_by_dir)

Funny that you say this now, just yesterday I finished the skeleton for https://github.com/adriangalilea/treedo

**Reply by jacostag** [reply]
@adriangalilea  looks really interesting, I am going follow it and try it

**Reply by adriangalilea** [reply]
> @adriangalilea looks really interesting, I am going follow it and try it

Won't be ready for a while, for now I'm thinking of the mental model.

Most concerned with the UX, then will cook it.

Many thanks.

Site should be ready later today.

**Reply by adriangalilea** [reply]
> @adriangalilea looks really interesting, I am going follow it and try it

Renamed the project and set up the site, after rename you may have lost star/follow no idea.

Namaste.

**Comment by jacostag** [comment]
@adriangalilea I think this ticket can be close, no?
